packages/better-php-doc-parser/src/AnnotationReader/NodeAnnotationReader.php

```diff
@@ -186,16 +204,45 @@ private function createPropertyReflectionFromPropertyNode(Property $property): ?
             // probably fresh node
             return null;
         }
-        if (! ClassExistenceStaticHelper::doesClassLikeExist($className)) {
+
+        if (! $this->reflectionProvider->hasClass($className)) {
             // probably fresh node
             return null;
         }
 
         try {
-            return new ReflectionProperty($className, $propertyName);
+            $classReflection = $this->reflectionProvider->getClass($className);
+            $propertyScope = $property->getAttribute(AttributeKey::SCOPE);
+            $propertyReflection = $classReflection->getProperty($propertyName, $propertyScope);
+
+            // @see https://github.com/phpstan/phpstan-src/commit/5fad625b7770b9c5beebb19ccc1a493839308fb4
+            return $this->privatesAccessor->getPrivateProperty($propertyReflection, 'reflection');
         } catch (Throwable $throwable) {
             // in case of PHPUnit property or just-added property
             return null;
         }
     }
+
+    private function resolveNativeClassMethodReflection(string $className, string $methodName): ReflectionMethod
+    {
+        if (! $this->reflectionProvider->hasClass($className)) {
+            throw new ShouldNotHappenException();
+        }
+
+        $reflectionClass = $this->reflectionProvider->getClass($className);
+        $methodReflection = $reflectionClass->getNativeMethod($methodName);
+
+        // @see https://github.com/phpstan/phpstan-src/commit/5fad625b7770b9c5beebb19ccc1a493839308fb4
+        $builtinMethodReflection = $this->privatesAccessor->getPrivateProperty($methodReflection, 'reflection');
+        if (! $builtinMethodReflection instanceof BuiltinMethodReflection) {
+            throw new ShouldNotHappenException();
+        }
+
+        $nativeMethodReflection = $builtinMethodReflection->getReflection();
+        if (! $nativeMethodReflection instanceof ReflectionMethod) {
+            throw new ShouldNotHappenException();
+        }
+
+        return $nativeMethodReflection;
+    }
```

Comment:
@ondrejmirtes I might got this wrong... :laughing:  is there any easier way to get native `ReflectionMethod`?

Reply:
@TomasVotruba This is the wrong way. I already told you: `$reflectionClass->getNativeReflection()->getMethod($methodName)` will get you an instanceof `\ReflectionMethod`.

Reply:
I didn't got that :) Thanks